SimCalorimetry/HGCalSimProducers/plugins/alpaka/HGCalDigiSoaFiller.cc

```diff
@@ -0,0 +1,185 @@
+#include "Geometry/HGCalMapping/interface/HGCalMappingTools.h"
+
+namespace ALPAKA_ACCELERATOR_NAMESPACE {
+
+  using namespace cms::alpakatools;
+
+  /**
+   *   @short consumes channel digis and packs then in a realistic FEDRawData stream
+  */
+  class HGCalDigiSoaFiller : public stream::EDProducer<> {
+  public:
+    //using CellIndexer = HGCalMappingCellIndexer;
+    using CellInfo = hgcal::HGCalMappingCellParamDevice;
+    using ModuleIndexer = HGCalMappingModuleIndexer;
+    using ModuleInfo = hgcal::HGCalMappingModuleParamDevice;
+
+    /**
+      @  short constructor
+    */
+    explicit HGCalDigiSoaFiller(edm::ParameterSet const& config);
+
+    /**
+       @short parameters to be used with this plugin
+    */
+    static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+
+    /**
+      @short steers the conversion to SOA
+    */
+    void produce(device::Event&, device::EventSetup const&) override;
+
+  private:
+    edm::EDGetTokenT<edm::SortedCollection<HGCalDataFrame> > digisCEET_, digisCEHSiT_, digisCEHSiPMT_;
+    device::EDPutToken<hgcaldigi::HGCalDigiDevice> digiProdT_;
+
+    edm::ESWatcher<HGCalElectronicsMappingRcd> cfgWatcher_;
+    //edm::ESGetToken<CellIndexer, HGCalElectronicsMappingRcd> cellIndexTkn_;
+    device::ESGetToken<CellInfo, HGCalElectronicsMappingRcd> cellTkn_;
+    edm::ESGetToken<ModuleIndexer, HGCalElectronicsMappingRcd> moduleIndexTkn_;
+    device::ESGetToken<ModuleInfo, HGCalElectronicsMappingRcd> moduleTkn_;
+  };
+
+  //
+  HGCalDigiSoaFiller::HGCalDigiSoaFiller(const edm::ParameterSet& config)
+      : digisCEET_(
+            consumes<edm::SortedCollection<HGCalDataFrame> >(config.getParameter<edm::InputTag>("eeChannelDigis"))),
+        digisCEHSiT_(
+            consumes<edm::SortedCollection<HGCalDataFrame> >(config.getParameter<edm::InputTag>("cehsiChannelDigis"))),
+        digisCEHSiPMT_(consumes<edm::SortedCollection<HGCalDataFrame> >(
+            config.getParameter<edm::InputTag>("cehsipmChannelDigis"))),
+        //cellIndexTkn_(esConsumes()),
+        cellTkn_(esConsumes()),
+        moduleIndexTkn_(esConsumes()),
+        moduleTkn_(esConsumes()) {
+    digiProdT_ = produces(config.getParameter<std::string>("digisProdName"));
+  }
+
+  //
+  void HGCalDigiSoaFiller::fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
+    edm::ParameterSetDescription desc;
+    desc.add<edm::InputTag>("eeChannelDigis", edm::InputTag("simHGCalUnsuppressedDigis", "EE"));
+    desc.add<edm::InputTag>("cehsiChannelDigis", edm::InputTag("simHGCalUnsuppressedDigis", "HEfront"));
+    desc.add<edm::InputTag>("cehsipmChannelDigis", edm::InputTag("simHGCalUnsuppressedDigis", "HEback"));
+    desc.add<std::string>("digisProdName", "HGC");
+    descriptions.addWithDefaultLabel(desc);
+  }
+
+  //
+  void HGCalDigiSoaFiller::produce(device::Event& iEvent, device::EventSetup const& iSetup) {
+    cfgWatcher_.check(iSetup);
```

Comment:
What is the intention with the `cfgWatcher`?

Reply:
This file was remove all together in https://github.com/CMS-HGCAL/cmssw/pull/137.